### PR TITLE
Utility to convert type proto to field type (if possible).

### PR DESCRIPTION
**Pull request by bgogul**

### CODE REVIEW

File: java/arcs/core/data/proto/TypeProtoDecoders.kt [hunk 1]
```diff
@@ -24,3 +26,26 @@ fun PrimitiveTypeProto.decode(): PrimitiveType =
...
+
+/**
+ * Converts a [TypeProto] protobuf instance into a Kotlin [FieldType] instance.
+ *
+ * @throws [IllegalArgumentexception] if the type cannot be converted to [FieldType].
+ */
+fun TypeProto.decodeAsFieldType(): FieldType =
+    when (getDataCase()) {
+        TypeProto.DataCase.PRIMITIVE -> getPrimitive().decodeAsFieldType()
+        // TODO: Handle FieldType.EntityRef. It is not clear how it is
```
Comment (bgogul) [inline java/arcs/core/data/proto/TypeProtoDecoders.kt]: @piotrswigon It is not clear how an `EntityRef` field is represented in the proto.

Comment (piotrswigon) [reply java/arcs/core/data/proto/TypeProtoDecoders.kt]: There is a reference type. You should be able to do TypeProto.DataCase.REFERENCE -> (...)

Comment (bgogul) [reply java/arcs/core/data/proto/TypeProtoDecoders.kt]: The `EntityRef` in Kotlin is defined as follows:
```
data class EntityRef(val schemaHash: String) : FieldType(Tag.EntityRef)
```
What is the  `schemaHash` value? It is not in the `recipe.proto` definition.

Comment (bgogul) [reply java/arcs/core/data/proto/TypeProtoDecoders.kt]: Summarizing the offline discussion here for the record. `schemaHash` is expected to be a cryptographic hash of the schema. Right now it is SHA-1 digest:  https://github.com/PolymerLabs/arcs/blob/1af1aac716359bda5bdee4b4f4ab62abfe3ad5e9/src/platform/digest-web.ts#L13

The schema is also normalized before it is hashed: 
https://github.com/PolymerLabs/arcs/blob/e587539077b35ef73bbbc46d67118231262a3b65/src/runtime/schema.ts#L243

We might want to save this in `recipe.proto`.

**Pull request merged**